frontend/src/services/GooglePlacesService.ts

```diff
@@ -0,0 +1,183 @@
+import axios, { AxiosError, AxiosResponse } from "axios";
+import * as dotenv from "dotenv";
+import { RestaurantDetails } from ".././types/RestaurantDetails";
+
+/* eslint-disable  no-console */
+// todo: remove this eslint-disable
+class GooglePlacesService {
+  public currRestaurants: Map<string, RestaurantDetails>;
+
+  constructor() {
+    dotenv.config();
+    this.currRestaurants = new Map();
+  }
+
+  getRestaurantPlaceID(
+    restaurantName: string,
+    restaurantLocation: string
+  ): Promise<string> {
+    const dataURI: string = encodeURIComponent(restaurantName);
+    let responseData: any[];
+
+    return axios
+      .get(
+        "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?",
+        {
+          params: {
+            input: dataURI,
+            inputtype: "textquery",
+            fields: "place_id",
+            key: process.env.REACT_APP_GOOGLE_PLACES_API_KEY,
+          },
+        }
+      )
+      .then((response: AxiosResponse) => {
+        if (response.data.candidates.length === 1) {
+          return Promise.resolve(response.data.candidates[0].place_id);
+        }
+        const promises: Array<Promise<boolean>> = [];
+        responseData = response.data.candidates;
```

Comment:
Also, can we make the `findplacefromtext` return the address as well? I think it can return something called `formatted_address` which may be the same as the `details` endpoint. Then we don't need to make another http request for each potential candidate.
![image](https://user-images.githubusercontent.com/49849754/103383712-9df56b80-4aa8-11eb-8de5-085f35ab3cea.png)